time-macros/src/error.rs

```diff
@@ -114,4 +114,30 @@ impl Error {
         .cloned()
         .collect()
     }
+
+    /// Like `to_compile_error`, but for use in macros that produce items.
+    pub(crate) fn to_compile_error_standalone(&self) -> TokenStream {
+        let (start, end) = (self.span_start(), self.span_end());
+
+        [
+            TokenTree::from(Punct::new(':', Spacing::Joint)).with_span(start),
+            TokenTree::from(Punct::new(':', Spacing::Alone)).with_span(start),
+            TokenTree::from(Ident::new("core", start)),
+            TokenTree::from(Punct::new(':', Spacing::Joint)).with_span(start),
+            TokenTree::from(Punct::new(':', Spacing::Alone)).with_span(start),
+            TokenTree::from(Ident::new("compile_error", start)),
+            TokenTree::from(Punct::new('!', Spacing::Alone)).with_span(start),
+            TokenTree::from(Group::new(
+                Delimiter::Parenthesis,
+                TokenStream::from(
+                    TokenTree::from(Literal::string(&self.to_string())).with_span(end),
+                ),
+            ))
+            .with_span(start),
+            TokenTree::from(Punct::new(';', Spacing::Alone)).with_span(end),
```

Comment:
The only difference is this semicolon, correct? Is there any reason we can't call `to_compile_error()` and then just append this single token?